maven-scm-providers/maven-scm-providers-git/maven-scm-provider-gitexe/src/main/java/org/apache/maven/scm/provider/git/gitexe/command/status/GitStatusConsumer.java

```diff
@@ -203,7 +203,7 @@ else if ( ( matcher = RENAMED_PATTERN.matcher( line ) ).find() )
         }
 
         // If the file isn't a file; don't add it.
-        if ( !files.isEmpty() && status != null )
+        if ( !files.isEmpty() )
```

Comment:
The else block does not cover status. Please explain.

Reply:
there is no options that status can be null. In the worst case will be exit in the else

` else
        {
            logger.warn( "Ignoring unrecognized line: " + line );
            return;
        }`

Reply:
In fact,  now double checking, `!files.isEmpty` is always true. files always have data

Reply:
Indeed!